File: src/main/resources/static/js/app/BankService.js
'use strict'
angular.module('demo.services', []).factory('BankService',
	["$http", "CONSTANTS", function ($http, CONSTANTS) {
		let service = {};
		service.getCities = function () {
			return $http.get(CONSTANTS.getCities);
		}
		service.findByCity = function (city) {
			return $http.get(`${CONSTANTS.findByCity}?city=${city}`, {cache: true});
		}
		return service;
	}]);

File: src/main/java/com/fyle/demo/service/BankService.java
package com.fyle.demo.service;

import com.fyle.demo.dto.GenericResponse;

public interface BankService {
  /**
   * find data By City
   * 
   * @param city
   * @return
   * @throws RuntimeException
   */
  GenericResponse findByCity(String city) throws RuntimeException;

  /**
   * get all Cities
   * 
   * @return
   * @throws RuntimeException
   */
  GenericResponse getCities() throws RuntimeException;
}
